### ISSUE TITLE
Bert Calculations Status

### ISSUE THREAD

**Issue by benfig1127** [benfig1127/Summer-Research-2020#9]
@mikeizbicki  I just finished writing all the code that we spoke about today. Everything does work and post correctly, however I am not sure that I did everything correctly. After writing my accuracy function 
```
import argparse
parser = argparse.ArgumentParser(fromfile_prefix_chars='@')


parser_data = parser.add_argument_group('data options')
parser_data.add_argument('--file1',type=str)
parser_data.add_argument('--file2',type=str)

args = parser.parse_args()

import torch
from statistics import mean

file1=torch.load(args.file1)
file2=torch.load(args.file2)

if file1 is None: 
	raise ValueError('Please specify which two embeddings to compare')
if file2 is None:
	raise ValueError('Please specify which two embeddings to compare')
	
accuracy=[]
embedding1_index=1
for embedding1 in file1:	
	dist_list=[]
	for embedding2 in file2: 
		dist=round(torch.dist(embedding1,embedding2,2).item(),4)
		dist_list.append(dist)
	min_dist=min(dist_list)
	min_dist_index=dist_list.index(min_dist)
	print('\n')

	if embedding1_index==min_dist_index:
		print('Success')
		accuracy.append(1)
	else:
		print('Failure')
		accuracy.append(0)
	print('Min Dist=',min_dist)
	print('File1 Index=',embedding1_index)
	print('Min Dist Index=',min_dist_index)
	print(dist_list)
	embedding1_index+=1
	final_score=mean(accuracy)
	score_output='Final Score {0:.0f}%' 
	print(score_output.format(final_score))
	print('\n')
```
I seem to be getting very poor accuracy despite only comparing 10 lines in both files. Here is the terminal output:
```
$ Summer_Work_mycomp python3 bert_accuracy.py --file1=/home/ben/Summer_Work_mycomp/test_UNv1.0.en-es.en_tensor_embedding --file2=/home/ben/Summer_Work_mycomp/test_UNv1.0.en-es.es_tensor_embedding

Failure
Min Dist= 1.7456
File1 Index= 1
Min Dist Index= 2
[2.1148, 1.8016, 1.7456, 1.779, 1.7606, 2.053, 1.7712, 1.814, 1.8891, 1.9513]
Final Score 0%

Failure
Min Dist= 1.6981
File1 Index= 2
Min Dist Index= 4
[2.1113, 1.7951, 1.7156, 1.7059, 1.6981, 2.036, 1.7374, 1.7273, 1.8619, 1.925]
Final Score 0%

Failure
Min Dist= 1.6131
File1 Index= 3
Min Dist Index= 2
[2.0364, 1.7, 1.6131, 1.6312, 1.6142, 1.9592, 1.6477, 1.6617, 1.793, 1.8456]
Final Score 0%

Failure
Min Dist= 1.6375
File1 Index= 4
Min Dist Index= 2
[2.0508, 1.723, 1.6375, 1.6503, 1.6484, 1.9836, 1.6991, 1.6746, 1.7984, 1.8637]
Final Score 0%

Failure
Min Dist= 1.6655
File1 Index= 5
Min Dist Index= 4
[2.0662, 1.7539, 1.6733, 1.6667, 1.6655, 2.002, 1.6997, 1.716, 1.8175, 1.8886]
Final Score 0%

Success
Min Dist= 1.9997
File1 Index= 6
Min Dist Index= 6
[2.3238, 2.1241, 2.0344, 2.0894, 2.0577, 2.271, 1.9997, 2.0783, 2.1615, 2.1265]
Final Score 0%

Failure
Min Dist= 1.5429
File1 Index= 7
Min Dist Index= 6
[1.9979, 1.6975, 1.6323, 1.6286, 1.6089, 1.9523, 1.5429, 1.6784, 1.7873, 1.8089]
Final Score 0%

Failure
Min Dist= 1.4372
File1 Index= 8
Min Dist Index= 6
[1.8364, 1.6277, 1.5475, 1.5546, 1.522, 1.8019, 1.4372, 1.5416, 1.6287, 1.6533]
Final Score 0%

Failure
Min Dist= 1.6489
File1 Index= 9
Min Dist Index= 6
[2.0079, 1.7787, 1.6886, 1.6875, 1.6559, 1.9731, 1.6489, 1.7056, 1.8222, 1.8506]
Final Score 0%

Failure
Min Dist= 1.79
File1 Index= 10
Min Dist Index= 6
[2.1024, 1.9638, 1.8657, 1.9056, 1.8703, 2.0884, 1.79, 1.8861, 1.9273, 1.9576]
Final Score 0%
```
Here is the code for the encoder, which looks like it was working to me, but perhaps you can give some insight.
```

class BertFineTuning(torch.nn.Module):
    def __init__(self):
        super().__init__()
        num_classes=100
        self.fc_class = torch.nn.Linear(768,num_classes)

    def forward(self,x):
        last_layer,embedding = bert(x) 
        embedding = torch.mean(last_layer,dim=1)
        out = self.fc_class(embedding)
        return out


if args.model == 'bert':
	model=BertFineTuning()
else:
	raise ValueError('Only bert model is supported right now')


# load the model
#checkpoint_name = 'bert-base-uncased'
checkpoint_name = 'bert-base-multilingual-uncased'
tokenizer = transformers.BertTokenizer.from_pretrained(checkpoint_name)
bert = transformers.BertModel.from_pretrained(checkpoint_name)

def bert_encodings(ss):
    input_ids=[]
    attention_masks=[]
    #encodings = []
    for line in ss:
        encoded_dict=tokenizer.encode_plus(line,add_special_tokens=True,max_length=64,pad_to_max_length=True,return_attention_mask=True,return_tensors='pt',)
        input_ids.append(encoded_dict['input_ids'])
        attention_masks.append(encoded_dict['attention_mask'])
    input_ids=torch.cat(input_ids,dim=0)
    attention_masks=torch.cat(attention_masks,dim=0)
    return input_ids
	

def data_parser():
	lines=[]
	with open(args.data,'r') as f_in:
		file_name=os.path.basename(args.data)+'_tensor_embedding'
		lines=[line.strip() for line in f_in]
			#lines.append(line.strip())
		line_tensor = bert_encodings(lines)
		embedding = model(line_tensor)	
		torch.save(embedding,file_name)	
		if args.print_tensor:
			print(embedding.shape)
```
I am also free whenever is convenient if you would like to try and work on this over zoom. Thanks

**Comment by mikeizbicki** [comment]
I'm on zoom right now if you want to drop in.
